domain/src/test/java/org/oppia/android/domain/exploration/ExplorationActiveTimeControllerTest.kt

```diff
@@ -0,0 +1,219 @@
+    val recordAggregateTimeProvider =
+      explorationActiveTimeController.recordAggregateTopicLearningTime(
+        profileId, TEST_TOPIC_ID_0, SESSION_LENGTH
+      )
+    monitorFactory.waitForNextSuccessfulResult(recordAggregateTimeProvider)
+  }
+
+  @Test
+  fun testSaveSessionLength_previousAggregateIsStale_overwritesOldAggregateWithCurrentLength() {
+    oppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    val currentTime = oppiaClock.getCurrentTimeMs()
+
+    // set aggregate learning time last updated timestamp to
+    // [LEARNING_TIME_STALENESS_THRESHOLD_MILLIS]+1 days ago.
+    val lastUpdatedTimeMs = currentTime.minus(LEARNING_TIME_STALENESS_THRESHOLD_MILLIS).minus(
+      TimeUnit.DAYS.toMillis(1)
+    )
+
+    createCacheStore(lastUpdatedTimeMs)
+
+    val recordAggregateTimeProvider =
+      explorationActiveTimeController.recordAggregateTopicLearningTime(
+        profileId, TEST_TOPIC_ID_0, SESSION_LENGTH
+      )
+
+    monitorFactory.waitForNextSuccessfulResult(recordAggregateTimeProvider)
+
+    val retrieveAggregateTimeProvider =
+      explorationActiveTimeController.retrieveAggregateTopicLearningTimeDataProvider(
+        profileId, TEST_TOPIC_ID_0
+      )
+
+    val aggregateTime = monitorFactory.waitForNextSuccessfulResult(retrieveAggregateTimeProvider)
+
+    assertThat(aggregateTime.lastUpdatedTimeMs).isEqualTo(currentTime)
+    assertThat(aggregateTime.topicLearningTimeMs).isEqualTo(SESSION_LENGTH)
+  }
+
+  @Test
+  fun testSaveSessionLength_previousAggregateIsNotStale_incrementsOldAggregateByCurrentLength() {
+    oppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    val currentTime = oppiaClock.getCurrentTimeMs()
+
+    // set aggregate learning time last updated timestamp to yesterday.
+    val lastUpdatedTimeMs = currentTime.minus(TimeUnit.DAYS.toMillis(1))
+
+    createCacheStore(lastUpdatedTimeMs)
+
+    val recordAggregateTimeProvider =
+      explorationActiveTimeController.recordAggregateTopicLearningTime(
+        profileId, TEST_TOPIC_ID_0, SESSION_LENGTH
+      )
+
+    monitorFactory.waitForNextSuccessfulResult(recordAggregateTimeProvider)
+
+    val retrieveAggregateTimeProvider =
+      explorationActiveTimeController.retrieveAggregateTopicLearningTimeDataProvider(
+        profileId, TEST_TOPIC_ID_0
+      )
+
+    val aggregateTime = monitorFactory.waitForNextSuccessfulResult(retrieveAggregateTimeProvider)
+
+    val expectedAggregate = SESSION_LENGTH + SESSION_LENGTH
+    assertThat(aggregateTime.lastUpdatedTimeMs).isEqualTo(currentTime)
+    assertThat(aggregateTime.topicLearningTimeMs).isEqualTo(expectedAggregate)
+  }
+
+  private fun createCacheStore(lastUpdatedTimeMs: Long) {
+    cacheFactory.createPerProfile(
+      CACHE_NAME,
+      createTopicLearningTimeDatabase(lastUpdatedTimeMs),
+      profileId
+    )
+  }
+
+  private fun createTopicLearningTimeDatabase(lastUpdatedTimeMs: Long): TopicLearningTimeDatabase {
```

Comment:
As discussed in an other comment, this probably isn't ideal since it leaks an implementation detail of the controller (using PersistentCacheStore), and tests are most robust when they focus on verifying a public API's contract.